api/tests/admin/custom_views/many_offers_operations_view_test.py

```diff
@@ -201,26 +205,32 @@ def test_get_current_criteria_on_active_offers(self):
         # Then
         assert result == expected_result
 
-    @pytest.mark.parametrize("already_set_incompatible", [False, True])
+    @pytest.mark.parametrize(
+        "validation_status",
+        [
+            OfferValidationStatus.DRAFT,
+            OfferValidationStatus.PENDING,
+            OfferValidationStatus.REJECTED,
+            OfferValidationStatus.APPROVED,
+        ],
+    )
     @patch("pcapi.core.search.async_index_offer_ids")
     @patch("wtforms.csrf.session.SessionCSRF.validate_csrf_token")
     def test_edit_product_gcu_compatibility(
-        self, mocked_validate_csrf_token, mocked_async_index_offer_ids, app, db_session, already_set_incompatible
+        self, mocked_validate_csrf_token, mocked_async_index_offer_ids, app, db_session, validation_status
     ):
         # Given
         users_factories.AdminFactory(email="admin@example.com")
         offerer = offerers_factories.OffererFactory()
         product_1 = offers_factories.ThingProductFactory(
             description="premier produit inapproprié",
             extraData={"isbn": "isbn-de-test"},
-            isGcuCompatible=not already_set_incompatible,
+            isGcuCompatible=not validation_status == OfferValidationStatus.REJECTED,
         )
         venue = offerers_factories.VenueFactory(managingOfferer=offerer)
-        offers_factories.OfferFactory(product=product_1, venue=venue, isActive=not already_set_incompatible)
+        offers_factories.OfferFactory(product=product_1, venue=venue, isActive=not validation_status)
```

Comment:
`not validation_status` n'est pas clair pour moi, comme `validation_status` est un enum

Reply:
bien vu, ça c'est une vraie erreur et le test passe probablement juste par un heureux hasard

Reply:
En fait le test passe car on a 2 offres à rejeter, alors que l'idée du test était de vérifier que ça marche bien quand on a déjà l'une des deux offres qui a précédemment été rejetée.
Donc je crois qu'on devrais passer `validation=REJECTED` (et la date et la raison) à `OfferFactory` ; on peut aussi vérifier qu'elle n'ait pas la même date et heure de rejet (quoique on doit du coup commencer avec un `freeze_time` pour espacer)

Reply:
bon en fait plus simplement je pense à : 
`offers_factories.OfferFactory(product=product_1, venue=venue, isActive=False, validation=offers_models.OfferValidationStatus.REJECTED, lastValidationType=OfferValidationType.CGU_INCOMPATIBLE_PRODUCT, lastValidationDate=datetime(2022, 8, 1))`

Reply:
en fait j'ai fait un dictionnaire `initial_offer_status` pour garder l'état des offres avant traitement ét vérifié que:
- `lastValidationDate` et `lastValidationType` ne changent pas dans le cas ou l'offre était déjà `REJECTED`
- `lastValidationDate` est récent et que `lastValidationType` est `CGU_INCOMPATIBLE_PRODUCT` sinon